Make Jotform integration functional

Niloth-p opened the pull request:
Fixes part of #32460.

Adds support for webhooks that send multipart/form-data payloads (Limitation: Files not supported yet).
Then updates the Jotform integration.
Updates to the message template and the screenshot are also included.

The 3 commits updating `webhooks/common.py`, `test_classes.py` and `generate-integration-docs-screenshot` can be squashd if necessary, as they have the common goal of supporting multipart/form-data in webhooks.

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented on zerver/lib/test_classes.py:
```diff
@@ -532,6 +532,9 @@ def client_post(
                 encoded = urlencode(info, doseq=True)
             else:
                 content_type = MULTIPART_CONTENT
+        if content_type.startswith("multipart/form-data"):
```
Should this be `elif`? Just for readability to make the control flow more obvious.

timabbott commented:
Looks good aside from a couple small comments.

Niloth-p commented:
@zulipbot add "integration review"

I've used '.multipart' as the file extension.

timabbott commented:
Looks great, merged, thanks @Niloth-p!

alya commented:
> Updates to the message template and the screenshot are also included.

I don't see screenshots of this. Please add them to the PR description and ping me for a review.

Niloth-p commented:
@alya 
> > Updates to the message template and the screenshot are also included.
> 
> I don't see screenshots of this. Please add them to the PR description and ping me for a review.

I meant the example screenshot, it's included in the file changes [here](https://github.com/zulip/zulip/pull/32481/files#diff-397fd1ed6cc4cc191d72aecaa31e80ab0c0156e9760c2e1c7298bb29085df60f), that was merged as part of the last commit.
I've now added them to the PR description as well.

I had previously proposed the italic version in #32413 (the commit wouldn't be visible now, because the branch was force pushed with a reset to exclude those commits before merging). After [this review comment](https://github.com/zulip/zulip/pull/32413#discussion_r1852699128) there on #32413, I raised a discussion in [CZO](https://chat.zulip.org/#narrow/channel/101-design/topic/Highlighting.20in.20messages.20generated.20by.20incoming.20webhooks/near/1987607) regarding the same (the italics Jotform image would be available here), and followed the [recommended](https://chat.zulip.org/#narrow/channel/101-design/topic/Highlighting.20in.20messages.20generated.20by.20incoming.20webhooks/near/1988875) format. I also [updated](https://chat.zulip.org/#narrow/channel/101-design/topic/Highlighting.20in.20messages.20generated.20by.20incoming.20webhooks/near/1990992) there that I've updated this PR with that format just in case any corrections were desired.

alya commented:
Sounds good. It's helpful to have a screenshot here even if it was also posted on CZO, as well as linking to the CZO thread from the PR description if/when you start one.

alya commented:
The screenshot is definitely an improvement, but we should aim to give examples that make sense for businesses for any software that is commonly used for business.

Niloth-p commented:
> It's helpful to have a screenshot here even if it was also posted on CZO, as well as linking to the CZO thread from the PR description if/when you start one.

That's true, sorry, I'll be more careful.

> The screenshot is definitely an improvement, but we should aim to give examples that make sense for businesses for any software that is commonly used for business. E.g., a shortened version of one of the templates on Jotform's website would probably work fine.

Thank you for the tip, @alya!
This was generated through a shortened Jotform template too, one for a tutoring business. But, I can understand your point, in terms of the relevance to businesses that use Zulip.

I shied away from forms with any technical terms, thinking of accessibility to people unfamiliar with the field of the business and/or the English terms. And preferred a use case that uses different data types (except links and datetime values) and form elements (since we're using the data as a test fixture). Hence, I thought this was a good fit.
I'll try to think of business use cases in the future.

Would this be a preferable alternative?

![001](https://github.com/user-attachments/assets/9f9078b9-c14d-47cd-8ca7-a1a861f331d6)

A fixture with question marks could be beneficial to test the formatting.
